cognite/neat/_graph/transformers/_rdfpath.py

```diff
@@ -90,20 +85,45 @@ def __init__(
 
         self.limit = limit
 
-    def transform(self, graph: Graph) -> None:
-        query = self._ref_template.format(
+    def _iterate_query(self) -> str:
+        query = """SELECT DISTINCT ?subject ?object
+                            WHERE {{
+                                ?subject a <{subject_type}> .
+                                ?subject <{subject_predicate}> ?value .
+                                ?object <{object_predicate}> ?value .
+                                ?object a <{object_type}> .
+                            }}"""
+
+        if self.limit and isinstance(self.limit, int) and self.limit > 0:
+            query += f" LIMIT {self.limit}"
+
+        return query.format(
             subject_type=self.subject_type,
             subject_predicate=self.subject_predicate,
             object_type=self.object_type,
             object_predicate=self.object_predicate,
         )
 
-        if self.limit and isinstance(self.limit, int) and self.limit > 0:
-            query += f" LIMIT {self.limit}"
+    def _count_query(self) -> str:
+        query = """SELECT (COUNT(DISTINCT (?subject ?object)) as ?count)
```

Comment:
If you set limit this will be invalid. I guess you have to add limit to this one as well.